From 1c3f230e6ee044fdcedccaae64bc5d3f341e5dec Mon Sep 17 00:00:00 2001
From: LuK1337 <priv.luk@gmail.com>
Date: Fri, 30 Aug 2024 01:34:30 +0200
Subject: [PATCH 4/4] oplus_dc_diming: improve dimlayer_bl_en getter

Now it returns proper values for v1/v3 oplus_dimlayer_bl_enable;

Change-Id: Ie14c39bb15dec633e454194e5c5df812d2ce114f
---
 techpack/display/oplus/oplus_dc_diming.c | 16 +++++++++++++++-
 1 file changed, 15 insertions(+), 1 deletion(-)

diff --git a/techpack/display/oplus/oplus_dc_diming.c b/techpack/display/oplus/oplus_dc_diming.c
index 86fb313adb2e..e876953d9886 100644
--- a/techpack/display/oplus/oplus_dc_diming.c
+++ b/techpack/display/oplus/oplus_dc_diming.c
@@ -1189,9 +1189,23 @@ int oplus_display_panel_set_dimlayer_enable(void *data)
 
 int oplus_display_panel_get_dimlayer_enable(void *data)
 {
+	struct dsi_display *display = NULL;
 	uint32_t *dimlayer_bl_enable = data;
 
-	(*dimlayer_bl_enable) = oplus_dimlayer_bl_enable_v2;
+	display = get_main_display();
+	if (!display)
+		return -EINVAL;
+
+	if (display->name) {
+		if (!strcmp(display->panel->oplus_priv.vendor_name, "ANA6706"))
+			(*dimlayer_bl_enable) = oplus_dimlayer_bl_enable;
+		else if (!strcmp(display->panel->name, "boe nt37800 amoled fhd+ panel with DSC"))
+			(*dimlayer_bl_enable) = oplus_dimlayer_bl_enable_v3;
+		else
+			(*dimlayer_bl_enable) = oplus_dimlayer_bl_enable_v2;
+	} else {
+		(*dimlayer_bl_enable) = oplus_dimlayer_bl_enable_v2;
+	}
 
 	return 0;
 }
-- 
2.49.0

